e2e_correctness/query_neo_mem.py

```diff
@@ -0,0 +1,275 @@
+    logger.debug(f"Memgraph json data {json_memgraph_data}")
+    graph = Graph()
+    vertices_id_mapings = {}
+    for item in json_memgraph_data:
+        if item["type"] == "node":
+            graph.add_vertex(extract_vertex_from_json(item))
+            vertices_id_mapings[item["id"]] = item["properties"]["id"]
+        else:
+            graph.add_edge(
+                create_edge_from_data(
+                    vertices_id_mapings[item["start"]],
+                    vertices_id_mapings[item["end"]],
+                    item,
+                )
+            )
+
+    graph.vertices.sort(key=lambda vertex: vertex.id)
+    graph.edges.sort(key=lambda edge: (edge.from_vertex, edge.to_vertex))
+    return graph
+
+
+def create_graph_neo4j_json(json_neo4j_data) -> Graph:
+    logger.debug(f"Memgraph json data {json_neo4j_data}")
+    graph = Graph()
+    vertices_id_mapings = {}
+    for item in json_neo4j_data:
+        if item["type"] == "node":
+            graph.add_vertex(extract_vertex_from_json(item))
+            vertices_id_mapings[item["id"]] = item["properties"]["id"]
+        else:
+            graph.add_edge(
+                create_edge_from_data(
+                    vertices_id_mapings[item["start"]["id"]],
+                    vertices_id_mapings[item["end"]["id"]],
+                    item,
+                )
+            )
+    graph.vertices.sort(key=lambda vertex: vertex.id)
+    graph.edges.sort(key=lambda edge: (edge.from_vertex, edge.to_vertex))
+    return graph
+
+
+def create_neo4j_driver(port: int) -> neo4j.BoltDriver:
+    return neo4j.GraphDatabase.driver(f"bolt://localhost:{port}", encrypted=False)
+
+
+def create_memgraph_db(port: int) -> gqlalchemy.Memgraph:
+    return gqlalchemy.Memgraph("localhost", port)
+
+
+def mg_execute_cyphers(input_cyphers: List[str], db: gqlalchemy.Memgraph):
+    """
+    Execute multiple cypher queries against Memgraph
+    """
+    for query in input_cyphers:
+        db.execute(query)
+
+
+def neo4j_execute_cyphers(input_cyphers: List[str], neo4j_driver: neo4j.BoltDriver):
+    """
+    Execute multiple cypher queries against Neo4j
+    """
+    with neo4j_driver.session() as session:
+        for text_query in input_cyphers:
+            query = neo4j.Query(text_query)
+            session.run(query).values()
+
+
+def run_memgraph_query(query: str, db: gqlalchemy.Memgraph):
+    """
+    Execute query against Memgraph
+    """
+    db.execute(query)
+
+
+def run_neo4j_query(query: str, neo4j_driver: neo4j.BoltDriver):
+    """
+    Execute query against Neo4j
+    """
+    with neo4j_driver.session() as session:
+        query = neo4j.Query(query)
+        session.run(query).values()
+
+
+def clean_memgraph_db(memgraph_db: gqlalchemy.Memgraph):
+    memgraph_db.drop_database()
+
+
+def clean_neo4j_db(neo4j_db: neo4j.BoltDriver):
+    with neo4j_db.session() as session:
+        query = neo4j.Query("MATCH (n) DETACH DELETE n;")
+        session.run(query).values()
```

Comment:
I think this function should remove other data such as indices, constraints and triggers as well.

Reply:
In e2e tests, we only clean data. I think we don't ever use inidces constraints and stuff like that. And since it is not online test I think it doesn't make sense to create triggers so I would leave that as is and extend later on

Reply:
Makes sense, this should be sufficient for now!